src/client/relatedFiles.ts

```diff
@@ -1,53 +1,94 @@
-    }
-
-    let relatedFile: string;
-    const fileName = editor.document.fileName;
-    const extension = path.extname(fileName).toLowerCase();
-    if (extension === '.html') {
-      const [tsFile, jsFile] = await Promise.all([
-        this.relatedFileExists(fileName, '.ts'),
-        this.relatedFileExists(fileName, '.js'),
-      ]);
-      if (tsFile) {
-        relatedFile = tsFile;
-      } else if (jsFile) {
-        relatedFile = jsFile;
-      }
-    } else if (extension === '.js' || extension === '.ts') {
-      relatedFile = await this.relatedFileExists(fileName, '.html');
-    }
-
-    if (relatedFile) {
-      commands.executeCommand('vscode.open', Uri.file(relatedFile), editor.viewColumn);
-    }
-  }
-
-  private async relatedFileExists(fullPath: string, relatedExt: string): Promise<string | undefined> {
-    const fileName = `${path.basename(fullPath, path.extname(fullPath))}${relatedExt}`;
-    fullPath = path.join(path.dirname(fullPath), fileName);
-
-    return new Promise<string | undefined>((resolve, reject) =>
-        fs.access(fullPath, fs.constants.R_OK, err => resolve(err ? undefined : fullPath)));
-  }
-}
+'use strict';
+import { commands, Disposable, TextEditor, TextEditorEdit, Uri, workspace } from 'vscode';
+import * as path from 'path';
+import * as fs from 'fs';
+import { AureliaConfigProperties } from './Model/AureliaConfigProperties';
+
+export class RelatedFiles implements Disposable {
+  private disposables: Disposable[] = [];
+
+  constructor() {
+    this.disposables.push(commands.registerTextEditorCommand('extension.auOpenRelated', this.onOpenRelated, this));
+
+    const fileExtensionsConfig = this.getFileExtensionsFromConfig();
+    const {
+      script: scriptExtension,
+      style: styleExtension,
+      unit: unitExtension,
+      view: viewExtension,
+    } = fileExtensionsConfig
+
+    this.disposables.push(commands.registerTextEditorCommand('extension.auOpenRelatedScript', this.openRelatedFactory(scriptExtension), this));
+    this.disposables.push(commands.registerTextEditorCommand('extension.auOpenRelatedStyle', this.openRelatedFactory(styleExtension), this));
+    this.disposables.push(commands.registerTextEditorCommand('extension.auOpenRelatedUnit', this.openRelatedFactory(unitExtension), this));
+    this.disposables.push(commands.registerTextEditorCommand('extension.auOpenRelatedView', this.openRelatedFactory(viewExtension), this));
+  }
+
+  public dispose() {
+    if (this.disposables.length) {
+      this.disposables.forEach((disposable) => {
+        disposable.dispose();
+      });
+    }
+  }
+
+  private getFileExtensionsFromConfig() {
+    const defaultSettings = {
+      script: '.js',
+      style: '.less',
+      unit: '.spec.js',
+      view: '.html',
+    };
+    return workspace.getConfiguration().get<AureliaConfigProperties['relatedFiles']>('aurelia.relatedFiles', defaultSettings);
+  }
+
+  private async onOpenRelated(editor: TextEditor, edit: TextEditorEdit) {

```

Comment:
> Changed logic from existing openRelated to get script extension from settings
(motivation: I believe an aurelia project will mostly have either only .js or .ts files)

Reply:
Not sure about this, but it might be more logical to remove the command 'open related aurelia file'. It feels a bit inconsistent together with the newly created commands that all specify what they will open. 

You need to remember now that is is basically an alias for open view or open script. However, it can be handy if you use it a lot because it will be on top of the command panel and then you don't need to switch between view and script in the list or it could be one keyboard combination if you bind it to that. 
With only the new commands available this would always require at least 2 key combinations, but then again you could also just create them for all 4 options, remember them and switch like a ninja between those 4, which might be more helpfull in the end.

In the following case the 'open related aurelia file' is a bit odd I think (was also the case in the current version, but less 'visible'):

Let's say you have a view-only component like

```html
<template bindable="foo,bar">
  <require from="boo.css" />
  <h1>${foo}</h1>
  <p>${bar}</p>
</template>
```
If I would now use the command 'open related aurelia file' would I then expect it to open the related CSS file, because the CSS file is the only related file to the HTML-only component :-) Instead I would need to use the open styles command.

How do you feel about removing the option 'open related aurelia file'?

---

Currently out of the box the command will only work in projects that use ```.js```, but I think we can fix that with the modification of the default settings to arrays.

Reply:
I also thought about removing it, but then again, this ninja switching is indeed handy.

How about renaming eg.
```diff
- open related aurelia file
+ switch between aurelia view and viewModel
```

As for the other possible combinations, I'd say we get the first version in
- 4 target switching
- (maybe renamed 'open related')
and get see the other as an enhancement for later?

Reply:
switch between aurelia view and viewModel 

sounds great. super idea; let's do that :-)